src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1301,6 +1301,8 @@ internal IBackendMessage SkipUntil(BackendMessageCode stopAt)
         internal async Task Rollback(bool async, CancellationToken cancellationToken = default)
         {
             Log.Debug("Rolling back transaction", Id);
+            // TODO: On successful rollback, the connector might be returned to the pool, if multiplexing is on.
```

Comment:
We should look into doing unbind as part of EndUserAction. I've had thoughts of linking the two mechanisms, but you're showing how important that is.